backends-common/rabbitmq/src/main/java/org/apache/james/backends/rabbitmq/ReactorRabbitMQChannelPool.java

```diff
@@ -159,10 +159,20 @@ private Channel tryBorrowFromPool() {
         return Mono.fromCallable(this::borrowFromPool)
             .doOnError(throwable -> LOGGER.warn("Cannot borrow channel", throwable))
             .retryBackoff(MAX_BORROW_RETRIES, MIN_BORROW_DELAY, FOREVER, Schedulers.elastic())
+            .onErrorMap(this::propagateException)
             .subscribeOn(Schedulers.elastic())
             .block();
     }
 
+    private Throwable propagateException(Throwable throwable) {
+        if (throwable instanceof IllegalStateException
```

Comment:
I guess we should open an issue on Reactor because IllegalStateException is not a meaningful type

Reply:
incoming https://github.com/reactor/reactor-addons/issues/223